megatron/training.py

```diff
@@ -69,9 +69,12 @@ def print_datetime(string):
 '''
 def _create_ds_config_dict():
     args = get_args()
-    with open(args.deepspeed_config, 'r', encoding='utf-8') as config_file:
-        ds_config_dict = json.load(config_file)
-    
+    if isinstance(args.deepspeed_config, dict) :
```

Comment:
Can you please share more on the scenario where `args.deepspeed_config` is a `dict`? Thanks!

Reply:
We are using the Ray + Megatron Deepspeed for distributed training. When I call the initialize_megatron() function, I pass the dict args of args_defaults as initialization value. Because it is distributed training, The configuration file does not exist on all machines. It is more convenient to pass in dict args from Ray.

Reply:
@yuanwu2017, awesome. Thanks for the response. 

I am generally curious about your Ray + Megatron DeepSpeed usage. Are you able to share any doc links? Thanks!

Reply:
Thanks @tjruwase . I will share with you later.

Reply:
Hi, @tjruwase  
The pretrain on Ray is opensource now.  You can  refer to the ReadME and source code. Currently it supports the two deferent pretrain solution. 
- Megatron-Deepspeed pretrain on Ray
- Megatron dataset + HF trainer + Deepspeed pretrain on Ray.

https://github.com/intel/llm-on-ray/tree/main/pretrain

Reply:
@yuanwu2017, thanks so much for sharing. This looks amazingly exciting. I will take a look. Congratulations on the release :)